pkg/devfile/parser/context/context.go

```diff
@@ -96,17 +92,6 @@ func (d *DevfileCtx) populateDevfile() (err error) {
 
 // Populate fills the DevfileCtx struct with relevant context info
 func (d *DevfileCtx) Populate() (err error) {
```

Comment:
should not remote this, this will cause regression to users who only pass in the context path to library.

Reply:
You mean users passing the parent directory containing the devfile(s)? Do you know which clients are passing a directory instead of the actual devfile path? The ones that I tested (`odo`, ODC) are parsing either from a file path (`odo`) or from in-memory bytes ([ODC](https://github.com/openshift/console/blob/master/pkg/devfile/handler.go#L55)).
And from the [`ParserArgs#Path`](https://github.com/devfile/library/blob/main/pkg/devfile/parser/parse.go#L83-L84) field description, it seems that we are expecting a relative or absolute path to the devfile path, right?
Anyway, just to be safe, I'll change that to make sure both cases are handled. Thanks!